plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php

```diff
@@ -40,19 +40,20 @@ public static function init() {
 
 		// Hook into WCA updated. This is hooked up here rather than in
 		// on_admin_init because that runs too late to hook into the action.
+		add_action( 'woocommerce_run_on_woocommerce_admin_updated' , array( __CLASS__, 'run_on_woocommerce_admin_updated' ) );
 		add_action(
 			'woocommerce_updated',
 			function() {
 				$next_hook = WC()->queue()->get_next(
 					'woocommerce_run_on_woocommerce_admin_updated',
-					array( __CLASS__, 'run_on_woocommerce_admin_updated' ),
+					array(),
 					'woocommerce-remote-inbox-engine'
 				);
 				if ( $next_hook === null ) {
 					WC()->queue()->schedule_single(
 						time(),
 						'woocommerce_run_on_woocommerce_admin_updated',
-						array( __CLASS__, 'run_on_woocommerce_admin_updated' ),
+						array(),
 						'woocommerce-remote-inbox-engine'
 					);
 				}
```

Comment:
🤔 Assuming the remote inbox engine is only initialized once, we might possibly be able to simplify this whole block of code to something like:

```
WC()->queue()->schedule_single(
 		time(),
 		'woocommerce_updated',
 		array( __CLASS__, 'run_on_woocommerce_admin_updated' ),
 		array(),
 		'woocommerce-remote-inbox-engine'
 );
 ```
 
 I think that shows the intent of the code more clearly and avoids needing the extra action (`woocommerce_run_on_woocommerce_admin_updated`).

Reply:
That's a good question!

Updated in https://github.com/woocommerce/woocommerce/pull/36768/commits/580b9ba069ccabe3936177fbf221b9d107c913aa

Reply:
Hey @adrianduffell ✋ 

Could you take a look at it one more time please?